public/js/controllers/address_controller.js

```diff
@@ -73,312 +104,672 @@ function plotGraph (processedData, otherOptions) {
     labelsKMB: true
   }
 
-  return new Dygraph(
-    document.getElementById('history-chart'),
-    processedData,
-    { ...commonOptions, ...otherOptions }
-  )
+  typesGraphOptions = {
+    labels: ['Date', 'Sending (regular)', 'Receiving (regular)', 'Tickets', 'Votes', 'Revocations'],
+    colors: ['#69D3F5', '#2971FF', '#41BF53', 'darkorange', '#FF0090'],
+    ylabel: 'Number of Transactions by Type',
+    title: 'Transactions Types',
+    visibility: [true, true, true, true, true],
+    legendFormatter: formatter,
+    plotter: barChartPlotter,
+    stackedGraph: true,
+    fillGraph: false
+  }
+
+  amountFlowGraphOptions = {
+    labels: ['Date', 'Received', 'Spent', 'Net Received', 'Net Spent'],
+    colors: ['#2971FF', '#2ED6A1', '#41BF53', '#FF0090'],
+    ylabel: 'Total Amount (DCR)',
+    title: 'Sent And Received',
+    visibility: [true, false, false, false],
+    legendFormatter: customizedFormatter,
+    plotter: barChartPlotter,
+    stackedGraph: true,
+    fillGraph: false
+  }
+
+  unspentGraphOptions = {
+    labels: ['Date', 'Unspent'],
+    colors: ['#41BF53'],
+    ylabel: 'Cummulative Unspent Amount (DCR)',
+    title: 'Total Unspent',
+    plotter: [Dygraph.Plotters.linePlotter, Dygraph.Plotters.fillPlotter],
+    legendFormatter: customizedFormatter,
+    stackedGraph: false,
+    visibility: [true],
+    fillGraph: true
+  }
+}
+
+function decodeZoom (encoded) {
+  if (!encoded) return false
+  var range = encoded.split('-')
+  if (range.length !== 2) {
+    return false
+  }
+  var start = parseInt(range[0], 36)
+  var end = parseInt(range[1], 36)
+  if (isNaN(start) || isNaN(end) || end - start <= 0) {
+    return false
+  }
+  return {
+    start: new Date(start * 1000),
+    end: new Date(end * 1000)
+  }
 }
 
 export default class extends Controller {
   static get targets () {
     return ['options', 'addr', 'btns', 'unspent',
-      'flow', 'zoom', 'interval', 'numUnconfirmed', 'formattedTime', 'txnCount']
+      'flow', 'zoom', 'interval', 'numUnconfirmed', 'formattedTime',
+      'pagesize', 'txntype', 'txnCount', 'qricon', 'qrimg', 'qrbox',
+      'paginator', 'pageplus', 'pageminus', 'listbox', 'table',
+      'range', 'chartbox', 'noconfirms', 'chart', 'pagebuttons']
   }
 
   initialize () {
```

Comment:
Most of the code inside `initialize` seems like it should be in `connect` instead so that for example if you navigate from one address to another via search bar, the state is wiped and rebuilt from scratch.
https://github.com/stimulusjs/stimulus/issues/75